### PR TITLE
refactor: move go packages into one, extract CLI cmd

### PR DESCRIPTION
**Pull request by nandor-magyar**
This could be the way to better organize code, just a quick attempt to do so.
@c3ppc3pp  @polaroi8d @m8vago  please have a quick look -- this is not the final version.

### CODE REVIEW

File: golang/Makefile [hunk 1]
```diff
@@ -1,16 +1,17 @@
-PACKAGE := github.com/dyrector-io/dyrectorio/agent
-CONTAINER_URL := ghcr.io/dyrector-io/dyrectorio/agent/
+PACKAGE := github.com/dyrector-io/dyrectorio/golang
+ORG_REGISTRY := ghcr.io/dyrector-io/dyrectorio/
```
Comment (polaroi8d) [inline golang/Makefile:2]: IMHO Put the slash end of the ENV is a bad small, the next line looks terrible.

File: golang/Makefile [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
+AGENT_REGISRY_URL := ${ORG_REGISTRY}agent/
 VCS_VERSION="$(shell git describe --exact-match --tags 2> /dev/null || git rev-parse --short HEAD)"
-AGENT_HASH="$(shell git rev-parse --short HEAD:../agent)"
+DYO_GOLANG_HASH="$(shell git rev-parse --short HEAD:../golang)"
```
Comment (polaroi8d) [inline golang/Makefile:5]: `DYO` is mandatory?

File: golang/Makefile [hunk 3]
```diff
@@ -67,18 +87,28 @@ security: # running gosec for static code analysis for bugs and leaks
 lint: # golangci for linting
 	golangci-lint run  ./... --timeout=3m
 
+.PHONY: build-cli
+build-cli: ##
+	cd build && \
+	docker build -t ${ORG_REGISTRY}cli:$(image_version) -f cli/Dockerfile . && \
```
Comment (polaroi8d) [inline golang/Makefile:93]: Same here, `${ORG_REGISTRY}/cli` would be better, IMHO

File: golang/Makefile (hunk 1 above)
Comment (nandor-magyar) [reply golang/Makefile:2]: These are not env varriables. I prefer that one too, even wondered why we did not opt for that initially. Replacing.

**Review by robot9706 — APPROVED** [review]
LGTM :octocat:

**Review by polaroi8d — APPROVED** [review]
Rebase please.

**Pull request merged**